spec/features/registration_spec.rb

```diff
@@ -18,33 +74,53 @@ def fill_in_fields(
     end
   end
 
-  # Click register Button
-  def click_register_button
-    click_button 'Cadastrar'
-  end
-
   before(:each) do
     visit register_url
     expect(page).to have_text 'Cadastro'
   end
 
   scenario 'registering with success' do
-    fill_in_fields
-    click_register_button
-    expect(page).to have_text 'register@test.com'
+    register_with_success
   end
 
   scenario 'registering without password should not pass' do
-    fill_in_fields EMAIL, '', ''
+    fill_in_user_fields EMAIL, '', ''
     click_register_button
     expect(page).to_not have_text EMAIL
     expect(page).to have_text 'Senha não pode ficar em branco'
   end
 
   scenario 'registering without password confirmation should not pass' do
-    fill_in_fields EMAIL, PASSWORD, ''
+    fill_in_user_fields EMAIL, PASSWORD, ''
     click_register_button
     expect(page).to_not have_text EMAIL
     expect(page).to have_text 'Confirmação de senha não é igual a Senha'
   end
+
+  context 'complete registration process' do
+    before { register_with_success }
+    scenario 'should see "Por favor, complete seu cadastro." message when registration is not completed' do
+      visit root_url
+      expect(page).to have_text('Por favor, complete seu cadastro.')
+    end
+
+    scenario 'should see validation messages on blank values' do
+      click_button 'Salvar'
+      expect(page).to have_content('8 erros ocorreram')
+      expect(page).to have_content('Nome não pode ficar em branco')
+      expect(page).to have_content('Sobrenome não pode ficar em branco')
+      expect(page).to have_content('Aniversário não pode ficar em branco')
+      expect(page).to have_content('Número de telefone não pode ficar em branco')
+      expect(page).to have_content('Endereço não pode ficar em branco')
+      expect(page).to have_content('Avatar não pode ficar em branco')
+      expect(page).to have_content('Nome do contato de emergência não pode ficar em branco')
+      expect(page).to have_content('Telefone do contato de emergência não pode ficar em branco')
+    end
+
+    scenario 'should complete registration' do
+      fill_persons_form person
+      click_button 'Salvar'
+      expect(page).to have_content(I18n.t('text.thanks.to_complete_account'))
```

Comment:
@marcoT89 Just check if the database was updated here too (since we don't know that yet through the UI)
You can use something like
 ``` ruby
expect(Person.last.emergency_contact.name).to eq(emergency_contact.name)
```
(The direct comparison will fail because emergency contact has no Id or person id, but the associated one has)